src/toil/jobStores/abstractJobStore.py

```diff
@@ -866,3 +884,28 @@ def _defaultTryCount(self):
     @classmethod
     def _validateSharedFileName(cls, sharedFileName):
         return bool(cls.sharedFileNameRegex.match(sharedFileName))
+
+
+class SupportJobStore(AbstractJobStore):
+    @classmethod
+    def _supportsUrl(cls, url, mode=mode.both):
+        def importOnly():
+            return url.scheme.lower() in ('http', 'https')
+
+        def exportOnly():
+            return False
+
+        def both():
+            return False
+
+        return cls._supportsUrlBody(mode, importOnly, exportOnly, both)
```

Comment:
The complexity you introduced stems from the fact that `both` implies `importOnly` and `exportOnly` (which, BTW, is why I didn't like the `Only` suffix). There are ways around this (e.g. using bitwise operators like in C), but again, we don't need `both`. 

`exportFile(…,'http://…')` should call `_supportsUrl('http://…',export=True)` on each job store class. None of them will return True and a specific exception should be raised, not `NotImplementedError`.